### PR TITLE
Self shield

### PR DESCRIPTION
**Pull request by FlanFlanagan**
@scopatz

### CODE REVIEW

**Comment by scopatz** [comment]
Hi @FlanFlanagan - can you please merge from develop so that you get the CI and test changes? Thanks!

File: pyne/ace.pyx [hunk 1]
```diff
@@ -300,7 +300,7 @@ class Library(object):
             tables_seen.add(name)
 
             # verify that we are suppossed to read this table in
-            if (table_names is not None) and (name not in table_names):
+            if (table_names is not None) and (name not in table_names):
```
Comment (scopatz) [inline pyne/ace.pyx:303]: Please do not add trailing whitespace. Thanks!

File: pyne/xs/cache.py [hunk 2]
```diff
@@ -60,7 +60,8 @@ def __init__(self, group_struct=None,
                                data_source.OpenMCDataSource,
                                data_source.SimpleDataSource,
                                data_source.EAFDataSource,
-                               data_source.NullDataSource,)):
+                               data_source.NullDataSource,),
+                 scalars=None):
```
Comment (scopatz) [inline pyne/xs/cache.py:64]: This should be indented to the same level as the line above.

File: pyne/xs/data_source.py [hunk 3]
```diff
@@ -886,7 +913,7 @@ class OpenMCDataSource(DataSource):
     stucture when the reactions are loaded in. Reseting this source group
     structure will clear the reaction cache.
     """
-
+
```
Comment (scopatz) [inline pyne/xs/data_source.py:916]: no trailing whitespace

File: pyne/xs/models.pyx [hunk 4]
```diff
@@ -367,6 +376,27 @@ def one_over_gamma_squared(E):
...
+    mask = (E < lower)
+    phi[mask] = np.sqrt(2*E[mask]*1e6) * 2 * np.pi * np.sqrt(E[mask]*1e6) 
+    phi[mask] *= np.exp(-E[mask]*1e6/(k*T)) / (np.pi * k * T)**1.5
+    mask = (E > lower)
+    phi[mask] = 1/((2*E[mask]*1e6)**0.5)
+    phi[mask] += 0.453 * np.exp(-1.036 * E[mask]) * np.sinh(np.sqrt(2.29 * E[mask]))
+    phi /= phi.sum()
+    return phi
+
+def fastspect(E, T=783, lower=1.0e-3):
```
Comment (scopatz) [inline pyne/xs/models.pyx:391]: PEP8: 2 lines between functions

File: pyne/xs/models.pyx [hunk 5]
```diff
@@ -367,6 +376,27 @@ def one_over_gamma_squared(E):
     return inv_g2
 
 
+def thermspect(E, T=573, lower=0.155e-6):
```
Comment (scopatz) [inline pyne/xs/models.pyx:379]: T should be a float and this function needs a docstring.

File: pyne/xs/models.pyx (hunk 4 above)
Comment (scopatz) [reply pyne/xs/models.pyx:391]: Also, T here should be a float and this function needs a docstring.

File: src/_atomic_data.cpp [hunk 6]
```diff
@@ -1,45 +1,25 @@
...
-      // that of index
-      if(pyne::nucname::znum(it->first) == z) {
-  // take atomic abundance and multiply by mass
-  // to get the mass of that nuclide / 100 since abundance is in %
-  element_atomic_weight += (it->second*atomic_mass_map[it->first]/100.0);
-      }
-    }
-    // insert the abundance of the element into the list
-    atomic_mass_map[z*10000000] = element_atomic_weight;
-  }
```
Comment (scopatz) [inline src/_atomic_data.cpp:39]: I think that changes to this file are not valid.  I am not sure why these sometimes get changed (I have noticed this on one of my machines, too). Please reset to the develop version.

File: pyne/xs/models.pyx (hunk 4 above)
Comment (scopatz) [reply pyne/xs/models.pyx:391]: This function has no unit tests

File: pyne/xs/models.pyx (hunk 5 above)
Comment (scopatz) [reply pyne/xs/models.pyx:379]: This function has no unit tests

**Comment by scopatz** [comment]
Thanks for the PR @FlanFlanagan! I think there are some relatively straightforward changes that need to happen, but overall looks great.

**Comment by scopatz** [comment]
haha whoops I didn't mean to close this. Also, I don't know why it isn't letting me reopen.

**Comment by FlanFlanagan** [comment]
No idea, I just assumed you did because of the rebase? Anyway, there is a new PR with the same branch so you can't reopen this one.

**Pull request closed without merging**